services/learn-card-network/brain-service/test/profiles.spec.ts

```diff
@@ -1698,4 +1492,112 @@ describe('Profiles', () => {
+                );
+            });
+
+            it('uses a supplied challenge if one is provided', async () => {
+                const result = await userA.clients.fullAuth.profile.generateInvite({ challenge: 'nice' });
+                expect(result.challenge).toEqual('nice');
+            });
+
+            it('does allow using multiple challenges', async () => {
+                await expect(
+                    userA.clients.fullAuth.profile.generateInvite({ challenge: 'c1' })
+                ).resolves.not.toThrow();
+                await expect(
+                    userA.clients.fullAuth.profile.generateInvite({ challenge: 'c2' })
+                ).resolves.not.toThrow();
+            });
+
+            it('does not allow using the same challenge more than once', async () => {
+                try {
+                    // Generate the first invite with the challenge 'nice'
+                    await userA.clients.fullAuth.profile.generateInvite({ challenge: 'nice' });
+                    console.log('First invite generated with challenge: "nice"');
+
+                    // Try to generate another invite with the same challenge 'nice'
+                    await userA.clients.fullAuth.profile.generateInvite({ challenge: 'nice' });
+                } catch (error) {
+                    console.log('Caught error:', error); // Log the actual error object
+                    expect(error).toMatchObject({
+                        code: 'CONFLICT',
+                        message: 'Challenge already in use!',
+                    });
+                }
+            });
+
+
+
+
+            it('allows setting the expiration date', async () => {
+                const result = await userA.clients.fullAuth.profile.generateInvite({ expiration: '24h' });
+                expect(result.expiresIn).toBe(24 * 60 * 60);
+            });
+        });
+
+        describe('connectWithInvite', () => {
+            it('allows users to connect via invite challenge', async () => {
+                const { challenge } = await userB.clients.fullAuth.profile.generateInvite();
+
+                await expect(
+                    userA.clients.fullAuth.profile.connectWithInvite({ profileId: 'userb', challenge })
+                ).resolves.toBe(true);
+
+                const oneConnection = await userA.clients.fullAuth.profile.connections();
+
+                expect(oneConnection).toHaveLength(1);
+                expect(oneConnection[0]!.profileId).toEqual('userb');
+            });
+
+            it('does not allow users to connect with an expired challenge', async () => {
+                const { challenge } = await userB.clients.fullAuth.profile.generateInvite({
+                    expiration: '1s',
+                });
+
+                console.log(`Test - Generated challenge: ${challenge}`);
+
+                await new Promise(resolve => setTimeout(resolve, 2000));
```

Comment:
Please use fake timers instead of arbitrarily waiting inside of tests and slowing them down!

https://vitest.dev/guide/mocking.html#timers

Reply:
thanks! *much* faster!